src/NWCClient.ts

```diff
@@ -895,6 +927,7 @@ export class NWCClient {
     resultValidator: (result: T) => boolean,
   ): Promise<(T & { dTag: string })[]> {
     await this._checkConnected();
+    await this._checkCompatibility();
```

Comment:
why did you choose to check compatibility here?

what about once in the constructor? then we also know what notification kind to subscribe to?

Reply:
It's an async function (fetching wallet service info), so I thought of doing the call before making a request

> then we also know what notification kind to subscribe to?

We are doing that right? We store the version in `this.version`/`this.supportedVersion` and use that before passing the kind in `subscribeNotifications`